File: TTTLive-OC/TTTLive/Settings/TTTSettingViewController.m
//
//  TTTSettingViewController.m
//  TTTLive
//
//  Created by yanzhen on 2018/9/13.
//  Copyright © 2018年 yanzhen. All rights reserved.
//

#import "TTTSettingViewController.h"
#import "TTTSCdnViewController.h"
#import "TTTSLocalViewController.h"

@interface TTTSettingViewController ()
@property (weak, nonatomic) IBOutlet UIView *cdnView;
@property (weak, nonatomic) IBOutlet UIView *localView;

@end

@implementation TTTSettingViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)settingChoiceAction:(UISegmentedControl *)sender {
    if (sender.selectedSegmentIndex == 0) {
        _localView.hidden = NO;
        _cdnView.hidden = YES;
        [self.view bringSubviewToFront:_localView];
    } else {
        _localView.hidden = YES;
        _cdnView.hidden = NO;
        [self.view bringSubviewToFront:_cdnView];
    }
}

- (IBAction)saveSettings:(id)sender {
    TTTSLocalViewController *localVc = nil;
    TTTSCdnViewController *cdnVc = nil;
    for (UIViewController *vc in self.childViewControllers) {
        if ([vc isKindOfClass:[TTTSLocalViewController class]]) {
            localVc = (TTTSLocalViewController *)vc;
        } else if ([vc isKindOfClass:[TTTSCdnViewController class]]) {
            cdnVc = (TTTSCdnViewController *)vc;
        }
    }
    NSString *error = [localVc saveSettings];
    if (error != nil) {
        [self showToast:error];
        return;
    }
    
    error = [cdnVc saveSettings];
    if (error != nil) {
        [self showToast:error];
        return;
    }
    TTManager.isCustom = YES;
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (IBAction)back:(id)sender {
    [self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: TTTLive/TTTLive/Settings/TTTSettingViewController.swift
//
//  TTTSettingViewController.swift
//  TTTLive
//
//  Created by yanzhen on 2018/9/11.
//  Copyright © 2018年 yanzhen. All rights reserved.
//

import UIKit

class TTTSettingViewController: UIViewController {

    @IBOutlet private weak var localView: UIView!
    @IBOutlet private weak var cdnView: UIView!
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    
    @IBAction func saveSettings(_ sender: Any) {
        var localVC: TTTSLocalViewController? = nil
        var cdnVc: TTTSCdnViewController? = nil
        for vc in childViewControllers {
            if vc is TTTSLocalViewController {
                localVC = vc as? TTTSLocalViewController
            } else if vc is TTTSCdnViewController {
                cdnVc = vc as? TTTSCdnViewController
            }
        }
        if let localError = localVC?.saveAction() {
            showToast(localError)
            return
        }
        
        if let cdnError = cdnVc?.saveAction() {
            showToast(cdnError)
            return
        }
        TTManager.isCustom = true
        dismiss(animated: true, completion: nil)
    }
    
    @IBAction func settingChoiceAction(_ sender: UISegmentedControl) {
        if sender.selectedSegmentIndex == 0 {
            localView.isHidden = false
            cdnView.isHidden = true
            view.bringSubview(toFront: localView)
        } else {
            localView.isHidden = true
            cdnView.isHidden = false
            view.bringSubview(toFront: cdnView)
        }
    }

    @IBAction func back(_ sender: Any) {
        dismiss(animated: true, completion: nil)
    }
    
}
